### ISSUE TITLE
Figure out some new ways to obtain parameters for XSS and SQLi modules - paramminer? or other ones?

### ISSUE THREAD

**Issue by kazet** [CERT-Polska/Artemis#1200]

**Comment by kshitijk4poor** [comment]
param miner is an extension for burp-site so idts that is possible to implement. there are still python/go options to work with tho

**Comment by kazet** [comment]
@dr4g0n369

**Comment by dr4g0n369** [comment]
Please assign this to me

**Comment by Fatima-Raees** [comment]
Hey @kazet , I'd like to work on this issue!

Before I start, I wanted to check which approach you'd prefer to go with first:

1. Arjun integration — subprocess call, JSON output, easy to wire into existing XSS/SQLi modules
2. x8 — faster (Rust-based), better for large-scale scans, slightly more complex to integrate
3. Passive first (ParamSpider/Wayback CDX) — zero requests to target, mines historical params
4. JS/HTML crawl parsing — extend the existing crawler to extract params from forms and JS files

Based on my research and taking help from AI I think we can start with Arjun as a lightweight enrichment step inside the existing XSS module's `process()` before spinning up a dedicated `param_discovery` Karton module — but happy to go whichever direction makes most sense for the project.

Also, is there a preferred way to emit discovered params downstream — new task type, or enriching the existing task payload?

Thanks!

**Comment by kazet** [comment]
Could you provide links and information what these tools take as input - do they process a single URL or something else?

To answer your second question: we already have a function that takes a url and returns parameter names: https://github.com/CERT-Polska/Artemis/blob/main/artemis/crawling.py#L38 - the goal would be to improve it ;)

**Comment by Fatima-Raees** [comment]
Thanks for the pointer @kazet !

I Checked the `crawling.py` and I think it makes sense to extend that function rather than build a separate module.

Here's what I'm thinking for improving it, in order of complexity:

1. Wayback CDX API : query `web.archive.org/cdx` for historically seen params on the domain. Fully passive (zero requests to target), pure Python, no new dependencies.
2. HTML form/input parsing : extract `name` attributes from forms, inputs, selects from the page response we're already fetching anyway.
3. Arjun (`--passive` mode) : pulls param names from CommonCrawl, OTX, and Wayback without touching the target. Could be a nice complement to number 1 option.- [Arjun](https://github.com/s0md3v/Arjun) — single URL. Pure Python, JSON output.
4. Arjun (active mode) :  brute-forces params against the live URL. More effective but heavier; could be behind a config flag.


My plan would be to start with number 1 and number 2 as they're low-risk and have no new binary dependencies, then discuss whether active discovery (number 4) is in scope.

Does that direction sound good to you? Happy to open a draft PR once we're aligned.

**Comment by kazet** [comment]
1 and 3 are a good direction, let's do it! 2 is already done but maybe can be improved?

**Comment by Fatima-Raees** [comment]
@kazet  Great, I'll work on 1 (Arjun passive)!

By looking at the current `get_injectable_parameters()` — it only extracts `<input name>` tags. Two quick improvements come to mind for 2 or with JS mining.
- Also cover `<textarea name>` and `<select name>` — same pattern, just not included yet
- Extract params already present in the URL itself (e.g. `?id=1` → `id`), since those are injectable too
- JavaScript source mining — scan inline <script> tags and linked .js files for patterns like:
URLSearchParams usage
fetch('/api?param=
$.ajax({ data: { param:
params.get('param')

Which one do you want me to start first,  the 2 improvement or 1 (Arjun passive). Kindly Assign this issue to me.

**Comment by kazet** [comment]
I think all make sense and any ordering would be good as every idea improves Artemis capability to identify parameters :)

**Comment by kazet** [comment]
Hello,

do you plan to work on an implementation?

**Comment by Fatima-Raees** [comment]
Yes I do, i am in the middle of my exams i wil do it after that is it okay? @kazet

**Comment by kazet** [comment]
Sure!

**Comment by elaliv** [comment]
Hi @kazet! As we discussed, I am ready to take this on.

**Comment by kazet** [comment]
Assigning to @elaliv due to inactivity
